### PR TITLE
feat: Intro to Mongoose and promises(#40010)

### PR DESCRIPTION
**Pull request by ajayrose25**

### CODE REVIEW

File: src/ controllers/trainee/routes.ts [hunk 1]
```diff
@@ -2,13 +2,14 @@ import { Router } from 'express';
...
+import { permissions } from '../../libs/routes/constants';
 const traineeRouter = Router();
 // authMiddleware('trainee', 'write'),
 
 traineeRouter.route('/')
-.get(validationHandler(validation.get), TraineeController.create)
+.get(authMiddleware(permissions.getUsers, 'read'), validationHandler(validation.get), TraineeController.create)
 .post(validationHandler(validation.create), TraineeController.list)
 .delete(TraineeController.delete)
 .put(validationHandler(validation.update), TraineeController.Update);
```
Comment (VisaaliSuccessive) [inline src/ controllers/trainee/routes.ts]: where is auth middleware ?

File: src/Server.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import * as bodyParser from 'body-parser';
 import Iconfig from './config/iconfig';
 import { errorHandler, notFoundRoute } from './libs/routes';
 import router from './router';
+import Database from './libs/database';
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: database folder name must be starting from capital

File: src/config/configuration.ts [hunk 3]
```diff
@@ -6,7 +6,8 @@ config();
 const configuration: Iconfig = Object.freeze( {
     port: process.env.PORT,
     env: process.env.NODE_ENV,
-    secretKey: process.env.SECRET_KEY
+    secretKey: process.env.SECRET_KEY,
+    mongoUri: process.env.MONGO_URL
```
Comment (VisaaliSuccessive) [inline src/config/configuration.ts:10]: Use destructuring

File: src/libs/database.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import * as mongoose from 'mongoose'
+import { promises, resolve } from 'dns';
+import { rejects } from 'assert';
```
Comment (VisaaliSuccessive) [inline src/libs/database.ts:3]: why are the packages imported? remove them

File: src/ controllers/trainee/routes.ts (hunk 1 above)
Comment (ajayrose25) [reply src/ controllers/trainee/routes.ts]: Done

File: src/Server.ts (hunk 2 above)
Comment (ajayrose25) [reply src/Server.ts]: Done

File: src/config/configuration.ts (hunk 3 above)
Comment (ajayrose25) [reply src/config/configuration.ts:10]: Done

File: src/libs/database.ts (hunk 4 above)
Comment (ajayrose25) [reply src/libs/database.ts:3]: Done

File: src/ controllers/trainee/routes.ts (hunk 1 above)
Comment (VisaaliSuccessive) [reply src/ controllers/trainee/routes.ts]: not done

File: src/config/configuration.ts (hunk 3 above)
Comment (VisaaliSuccessive) [reply src/config/configuration.ts:10]: not done

**Comment by VisaaliSuccessive** [comment]
done changes in other PR so closing this PR without merging this

**Pull request closed without merging**